azure_translator/errors.py

```diff
@@ -0,0 +1,41 @@
+"""Azure errors module."""
+import xml.etree.ElementTree as ET
+
+
+class BaseAzureException(Exception):
+    """
+    Base exception for the SDK.
+    """
+    def __init__(self, *args, **kwargs):
+        self.response = kwargs.pop('response', None)
+        self.request = kwargs.pop('request', None)
+        super(BaseAzureException, self).__init__(*args, **kwargs)
+
+
+class AzureApiError(BaseAzureException):
+    """
+    Raised when the API returns a non-200 body.
+    """
+    MSG_SEPARATOR = "; "
+
+    def __init__(self, msg, *args, **kwargs):
+        response = kwargs.get('response')
+        if response:
+            try:
+                msg = self.MSG_SEPARATOR.join(ET.fromstring(response.content).itertext())
+            except ET.ParseError:
+                pass
+
+        super(AzureApiError, self).__init__(msg, *args, **kwargs)
+
+
+class AzureApiBadFormatError(BaseAzureException):
```

Comment:
Nevermind, I initially confused that naming with the "Argument Exception" we're dealing with here :white_check_mark: